src/processors/tone/ladder_filter/LadderFilterProcessor.cpp

```diff
@@ -0,0 +1,135 @@
+#include "../../ParameterHelpers.h"
+
+#include "LadderFilterProcessor.h"
+
+LadderFilterProcessor::LadderFilterProcessor (UndoManager* um) : BaseProcessor ("Ladder Filter", LadderParameters::createParameterLayout(), um), p (vts)
+{
+    // Info and style
+    uiOptions.info.description = "Resonant 4-pole ladder high-pass and low-pass,\n capable of self-oscillation. With nonlinear drive.";
+    uiOptions.info.authors = StringArray { "Butch Warns" };
+    uiOptions.info.infoLink = "https://butchwarns.de/";
+    uiOptions.backgroundColour = juce::Colour::fromRGB (0xBB, 0xE1, 0xC3);
+    uiOptions.powerColour = juce::Colour::fromRGB (0x99, 0x0B, 0x3C);
+
+    addPopupMenuParameter ("FILTER_MODE");
+}
+
+void LadderFilterProcessor::prepare (double sampleRate, int samplesPerBlock)
+{
+    typedef LadderParameters params;
+    drive_smooth.mappingFunction = params::drive;
+    hp_cutoff_smooth.mappingFunction = params::hp_cutoff;
+    hp_resonance_smooth.mappingFunction = params::hp_resonance;
+    lp_cutoff_smooth.mappingFunction = params::lp_cutoff;
+    lp_resonance_smooth.mappingFunction = params::lp_resonance;
+
+    drive_smooth.setParameterHandle (p.drive_norm);
+    drive_norm_smooth.setParameterHandle (p.drive_norm);
+    hp_cutoff_smooth.setParameterHandle (p.hp_cutoff_norm);
+    hp_resonance_smooth.setParameterHandle (p.hp_resonance_norm);
+    lp_cutoff_smooth.setParameterHandle (p.lp_cutoff_norm);
+    lp_resonance_smooth.setParameterHandle (p.lp_resonance_norm);
+
+    drive_smooth.prepare (sampleRate, samplesPerBlock);
+    drive_norm_smooth.prepare (sampleRate, samplesPerBlock);
+    hp_cutoff_smooth.prepare (sampleRate, samplesPerBlock);
+    hp_resonance_smooth.prepare (sampleRate, samplesPerBlock);
+    lp_cutoff_smooth.prepare (sampleRate, samplesPerBlock);
+    lp_resonance_smooth.prepare (sampleRate, samplesPerBlock);
+
```

Comment:
For the resonance parameters, I wonder if we might also want to set the mapping functions to handle the "skew" operation in the smoother, instead of in the `filter.set_resonance()` call? This would mean that when the parameter is being smoothed, we would only need to call the skew function once per-block instead of once per-sample.

Basically we would do:
```cpp
hp_resonance_smooth.mappingFunction = [] (float val)
{
    return ladder_filter_utility::map_linear_normalized (val, 0.0, 4.0);
};
```

And then edit the `filter.set_resonance()` functions so they don't apply the skew a second time.

Reply:
Actually, this would allow us to implement the Filter Mode parameter as well:
```cpp
hp_resonance_smooth.mappingFunction = [this] (float val)
{
    if (p.filter_mode_norm == 0.0f)
        val *= 0.5f; // less resonance in "normalized" mode
    return ladder_filter_utility::map_linear_normalized (val, 0.0, 4.0);
};
```
Or something like that... you probably have a better idea of how you would want that to work.

Reply:
And one more thing... I think we'll need a mapping function for `drive_smooth` also, since if I remember right, that parameter needed some kind of Decibel conversion to end up in the correct range?

Reply:
Hi Jatin, many thanks for your ideas on this! 

The resonance skew was already done in the smoother, but I moved the final denormalization call there too, which makes sense of course. 

I also moved the skew/denormalization functions out of the `LadderParameters` class and instead use lambdas now, as per your suggestion. The problem I was encountering before was the dependency of the resonance mapping on the `p.filter_mode_norm` value, which is obviously not working in a static method. Filter mode selection now works well.

The only other issue I was having was my unsuccessful attempt of making the `filter_mode_norm` anything other than a `std::atomic<float>*`, as this always resulted in an immediate crash when trying to access the value. If you feel this needs changing, please tell me. Any hint on how to use a `chowdsp::ChoiceParameter` here would be welcome in that case :)

Anyway, the processor seems to work fine now. Feel free to do your own checks, of course!
Thanks again.

-Butch